Compiler/BackEnd/Tearing.mo

```diff
@@ -3297,50 +3276,35 @@ algorithm
   end matchcontinue;
 end TarjanGetAssignable;
 
-
-protected function getUnassigned " finds the unassigned vars or eqs.combine with List.fold"
-  input Integer assEntry;
-  input tuple<Integer,list<Integer>> InValue;
-  output tuple<Integer,list<Integer>> OutValue;
-algorithm
-OutValue := match(assEntry,InValue)
-  local
-    Integer indx;
-    list<Integer> lst;
-  case(_,(indx,lst))
-    then
-      if intEq(assEntry,-1) then
-        ((indx+1,indx::lst))
-      else
-        ((indx+1,lst));
-  end match;
-end getUnassigned;
-
-
 protected function traverseEqnsforAssignable
 " selects next equations that can be causalized
   author: ptaeuber FHB 2013-10"
-  input list<Integer> inAssEq;
+  input array<Integer> inAss;
   input BackendDAE.IncidenceMatrix m;
   input array<list<Integer>> mapEqnIncRow;
   input array<Integer> mapIncRowEqn;
   input Integer prescient;
-  output list<Integer> outAssEq_multi = {};
-  output list<Integer> outAssEq_single = {};
+  output list<Integer> selectedrows;
 protected
   Integer eqnColl,eqnSize;
+  DoubleEndedList<Integer> delst;
 algorithm
-  for e in inAssEq loop
+  delst := DoubleEndedList.empty(0);
+  for e in 1:arrayLength(inAss) loop
+    if arrayGet(inAss,e)<>-1 then
+      continue;
+    end if;
     eqnColl := mapIncRowEqn[e];
     eqnSize := listLength(mapEqnIncRow[eqnColl]);
     if listLength(m[e]) == eqnSize + prescient then
       if eqnSize == 1 then
-        outAssEq_single := e::outAssEq_single;
+        DoubleEndedList.push_back(delst, e);
       else
-        outAssEq_multi := e::outAssEq_multi;
+        DoubleEndedList.push_front(delst, e);
```

Comment:
The difference in results should be due to this line: it reverses the old order of the queue, and I think the changed result is just due to randomness in the algorithm. @vruge @ptaeuber do you think the order of elements in outAssEq_multi matters?